datafusion/core/src/execution/runtime_env.rs

```diff
@@ -152,7 +153,16 @@ pub struct RuntimeConfig {
 impl RuntimeConfig {
     /// New with default values
     pub fn new() -> Self {
-        Default::default()
+        let mut table_factories: HashMap<String, Arc<dyn TableProviderFactory>> =
+            HashMap::new();
+        table_factories.insert("PARQUET".into(), Arc::new(ListingTableFactory::new()));
+        table_factories.insert("CSV".into(), Arc::new(ListingTableFactory::new()));
+        table_factories.insert("JSON".into(), Arc::new(ListingTableFactory::new()));
```

Comment:
do we need to add an entry for "NDJSON" as well? See https://github.com/apache/arrow-datafusion/issues/4198

Reply:
> do we need to add an entry for "NDJSON" as well? See #4198

This would be possible, but currently (logical/datasource) FileType::from_str only knows about avro,parquet,json and csv and will err.. So more changes will be needed to enable NDJSON again..

Reply:
This, NDJSON missing, and compression only being supported in some code paths is the reason I believe this refactoring is much needed...